### PR TITLE
Implement logic for equalities involving the same variable twice

### PR DESCRIPTION
**Pull request by gwils**
Please review the following code carefully. I've written tests, but I'm not sure it's correct.

### CODE REVIEW

File: src/Domain/Internal.hs [hunk 1]
```diff
@@ -279,9 +285,34 @@ eqRef x d y = do
...
+  if xroot == yroot then do
+    let zlop = no u ylop yhip
+        zhip = no u yhip ylop
+        (todo,  xlo') = maxx zlop xlop t (no u ylo yhi) xlo
+        (todo', xhi') = minx zhip xhip t (no u yhi ylo) xhi
+    case u of
+      -- eg. x = -x+4 --> x must be 2
+      NegativeOne -> do
+        let solution = k `div` 2
+            result = mkR (yrank+1) (Right solution) (Right solution) nil nil nil
```
Comment (gwils) [inline src/Domain/Internal.hs]: Is this rank correct?

File: src/Domain/Internal.hs [hunk 2]
```diff
@@ -279,9 +285,34 @@ eqRef x d y = do
...
+        guard $ even k && inside xlo' xhi' solution
+        writeRef xroot $ Root result
+        runPs (todo <> todo') xroot
+        runKs (rel t ycov <> xcov) solution
+      -- x = x+c --> c must be zero
+      One -> do
+        guard $ k == 0
+        let result = mkR (yrank+1) xlo' xhi' (xlop <> rel t zlop)
```
Comment (gwils) [inline src/Domain/Internal.hs]: Is this rank correct?

**Comment by gwils** [comment]
ping @ekmett

**Comment by ekmett** [comment]
Merging. Any issues will be resolved post merge.

**Pull request merged**